packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -97,7 +103,13 @@ module.exports = ({ strapi }) => ({
 
     const entities = await strapi.entityService.findPage(uid, params);
 
-    const mappedResults = await mapAsync(entities.results, (entity) => this.mapEntity(entity, uid));
+    if (!entities) {
+      return entities;
+    }
+
+    const mappedResults = await mapAsync(entities.results || [entities], (entity) =>
+      this.mapEntity(entity, uid)
+    );
     return { ...entities, results: mappedResults };
```

Comment:
we shouldn't return results if it's a single type no? maybe we have to differentiate this logic if it's a single type or not 🤔